build/tasks/docs.js

```diff
@@ -6,45 +6,57 @@ const glob = require('glob');
+const REGEX_FILE_NAME = /((\w)+(-)*(\w))+(?=.js)/g;
 
+function createDocForFile(filename) {
+  const splittedName = filename.split('/');
+  const filenameWithoutExt = filename.match(REGEX_FILE_NAME)[0];
 
+  return docjs.build([filename], {shallow: true})
+    .then(result => docjs.formats.md(result, {}))
+    .then(output => ({
+      file: filenameWithoutExt,
+      // get the parent folder of file
+      folder: filenameWithoutExt !== 'index' ? splittedName[splittedName.length - 2] : 'index',
+      markdown: output
+    }));
+}
 
 // deletes all files in the output path
 gulp.task('clean-docs', () =>
   gulp.src([`${paths.docsOutput}`])
     .pipe(vinylPaths(del))
 );
 
-gulp.task('build-jsdoc-to-md', () => {
-  const files = glob.sync(paths.source); // glob allows pattern matching for filenames
-  const regexFileName = /((\w)+(\-)*(\w))+(?=.js)/g;
-  const createdMdDocs = [];
-  files
-  .filter(filename => {
-    const name = filename.match(regexFileName)[0];
-    return name !== 'imports';
-  })
-  .map(filename => {
-    const splittedName = filename.split('/');
-    const filenameWithoutExt = filename.match(regexFileName)[0];
-    const jsDocObj = docjs.buildSync([filename], {shallow: true});
-    return docjs.formats.md(jsDocObj, {}, (err, output) => {
-      createdMdDocs.push({
-        file: filenameWithoutExt,
-        folder: filenameWithoutExt !== 'index' ? splittedName[splittedName.length - 2] : '',  // get the parent folder of file
-        markdown: output
-      });
-    });
-  });
-  const writeMdFiles = createdMdDocs.map(mdDoc => {
-    const folder = `${paths.docsOutput}${mdDoc.folder}`;
-    const filePath = mdDoc.file !== 'index' ? `${paths.docsOutput}${mdDoc.folder}/${mdDoc.file}.md` : `${paths.docsOutput}${mdDoc.file}.md`;
-    if (!fs.existsSync(folder)) {
-      fs.mkdirSync(folder);
-    }
-    return fs.writeFile(filePath, mdDoc.markdown);
-  });
-  return Promise.all(writeMdFiles);
+gulp.task('build-jsdoc-to-md', done => {
+  Promise.resolve(glob.sync(paths.source))
+    // Filter file names to match tag name pattern
+    .then(fileNames => fileNames.filter(fileName => {
+      const name = fileName.match(REGEX_FILE_NAME)[0];
+      return name !== 'imports' && name !== 'index';
+    }))
+    // Generate markdown for each file
+    .then(fileNames =>
+      Promise.all(fileNames.map(createDocForFile))
+    )
+    // Group the markdown docs by there folder
+    .then(docs => docs.reduce((groups, doc) => {
+      groups[doc.folder] = groups[doc.folder] || [];
+      groups[doc.folder].push(doc);
+      return groups;
+    }, {}))
+    // Create markdown files for each folder
+    .then(groups => Object.keys(groups).forEach(folderName => {
+      const group = groups[folderName];
+      const fileName = `${paths.apiDoc}/${folderName}.md`;
+      const normalize = doc => doc.markdown.replace(/<!--.*?-->\n*/g, '');
+      const content = `# ${folderName}\n${group.map(normalize).join('')}`;
```

Comment:
I think this is related to how the API documentation is later included in the styleguide.
I remember that @fragsalat told me that his markdown-interpreter for the styleguide needs this "#" which is basically markdown headline (h1) :D